### PR TITLE
Update models.py

### PR DESCRIPTION
**Pull request by plotxy**
If abv is string (often is '-') code will fail. 
You can use 'Beer/ShowBrewerBeers.asp?BrewerID=233' for testing.

### CODE REVIEW

**Comment by parryc** [comment]
@plotxy what is the purpose of the additional encode step?

**Comment by plotxy** [comment]
Using the get_beers function, if in the page there is no ABV but a "-" you get an error because it can't convert a string to float. My add makes sure that ABV is not string.

File: ratebeer/soup.py [hunk 1]
```diff
@@ -41,4 +41,4 @@ def _get_soup(url):
     req = requests.get(_BASE_URL + url, allow_redirects=True)
     if "ratebeer robot oops" in req.text.lower():
         raise rb_exceptions.PageNotFound(url)
-    return BeautifulSoup(req.text, "lxml")
+    return BeautifulSoup(req.text.encode('utf-8','Windows-1252'), "lxml")
```
Comment (parryc) [inline ratebeer/soup.py]: @plotxy sorry to be more precise, what is the purpose of changing the encoding here?

**Comment by parryc** [comment]
@plotxy I was referring to the encoding change. I added a comment on the line in question.

**Comment by plotxy** [comment]
Oh damn, I'm so sorry. That wasn't supposed to be there.
Long story short: I had encoding issues because lxml didn't work properly on my computer and was using html5lib, but apparently doesn't go smooth. That fixed the problem, but it was supposed to be for my own only. Sorry.

**Comment by plotxy** [comment]
BTW, I realized my solution is not Python3 compatible, I tried to use the str type, but that's not compatible with Python2.7.

A solution I found is
`from six import string_types
if abv and not isinstance(abv, string_types):
                beer.abv = float(abv)`

**Comment by parryc** [comment]
No worries – it looks like you're merging the head of your master branch, which is why it picked up your encoding change.  I'm not good enough at git to say one way or the other, but it might be worth rolling back to the original commit for the pull request and then adding the `string_types` solution you have above. 

The `six` solution looks good to me! I'd suggest adding `six` as an explicit dependency in the requirements.txt file, as well. 

@alilja, I think the ball's in your court now.

**Comment by OrganicIrradiation** [comment]
It would be nice to avoid adding another dependency (i.e., ```six```) and there seem to be a number of ways to test for this condition, so I would vote for an alternative solution.

**Comment by parryc** [comment]
I thought one of our existing dependencies had a dependency on six, but I just rechecked with `pip show` and `six` isn't a part of any of them. 

Could just throw it in a try/except.

**Comment by OrganicIrradiation** [comment]
Just to confirm, this fails:

``` python
from ratebeer import RateBeer
from pprint import pprint

rb = RateBeer()
b = rb.get_brewery('/brewers/deschutes-brewery/233/')
pprint([i.__dict__ for i in b.get_beers()])
```

While the beer with no ABV works when queried directly:

``` python
b = rb.get_beer('/beer/deschutes-altitude-amber/92102/')
pprint(b._populate().__dict__)

{'_has_fetched': True,
 'abv': None,
 'brewed_at': None,
 'brewery': <Brewery('/brewers/deschutes-brewery/233/')>,
 'calories': None,
 'description': 'A high-powered version of a typical amber',
 'ibu': None,
 'img_url': 'https://res.cloudinary.com/ratebeer/image/upload/w_120,c_limit/beer_92102.jpg',
 'mean_rating': 3.75,
 'name': 'Deschutes Altitude Amber',
 'num_ratings': 2,
 'overall_rating': None,
 'retired': False,
 'seasonal': 'Special',
 'style': 'Amber Ale',
 'style_rating': None,
 'style_url': '/beerstyles/amber-ale/53/',
 'tags': [],
 'url': '/beer/deschutes-altitude-amber/92102/',
 'weighted_avg': 3.3}
```

Perhaps replacing the attribute population with try/excepts similar to the Beer class would work? (untested) It seems a bit verbose and silently "fails", but I'm not sure if there's a preferred way of dealing with these possibilities.

```python
try:
    beer.abv = float(row.findAll('td')[1].text)
except ValueError: # Empty ABV: '-'
    beer.abv = None
try:
    beer.weighted_avg = float(row.findAll('td')[4].text.strip())
except ValueError:
    beer.weighted_avg = None
try:
    beer.style_rating = int(row.findAll('td')[5].text.strip())
except ValueError:
    beer.style_rating = None
try:
    beer.num_ratings = int(row.findAll('td')[6].text.strip())
except ValueError:
    beer.num_ratings = None
```

**Comment by parryc** [comment]
Helper class, perhaps?

```
def _cast(type, row, idx):
  try:
    if(type == 'float'):
      return float(row.findAll('td')[idx].text.strip())
    ...
  except:
    return None

beer.adv = _cast('int', row, 1)
```

**Comment by OrganicIrradiation** [comment]
That sounds good to me! @plotxy, want to take a shot at it?

**Comment by parryc** [comment]
Closing this pull request because RateBeer has switched to a new UI (again! See: #48) and thus the above is no longer relevant.

**Pull request closed without merging**